### DISCUSSION TITLE
Rotordynamic

### DISCUSSION THREAD

**Discussion by Ayaa30** [OpenFAST/openfast#1195] [General]
Hello @jjonkman  

I am Aya working on my master of science in mechanical studying rotordyanmic of wind turbine drivetrain. I am new to OpenFast and I was wondering if I can obtain the natural frequencies of the turbine drive shaft added to it the masses of the assembled components with their inertias. Thanks in advance 

Best Regards
Aya

**Comment by jjonkman** [comment]
Dear @Ayaa30,

I'm sorry, but I'm not sure I understand your question.  Please clarify.

Best regards,

**Reply by Ayaa30** [reply]
Dear JJonkman

I need to determine the eigenvalues of a turbine drivetrain  and find its mode shapes.

Best Regards,

**Reply by jjonkman** [reply]
Dear @Ayaa30,

Well, the drivetrain model in OpenFAST is currently quite simplified, accounting for the rigid body rotation of the generator relative to the nacelle and a torsional spring-damper element between the generator and hub representing the torsional behavior of the drivetrain.  The generator, hub, and gears are rigid (with a fixed gearbox ratio).  For a rigid rotor and support structure, analytical expressions exist for the drivetrain natural frequency based on the torsional spring stiffness and inertias of the generator and rotor.  For a flexible rotor and support structure, the linearization functionality of OpenFAST can be used to calculate, the natural frequency of the drivetrain, including coupling to the flexible rotor and support structure.  

Best regards,
